libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -61,6 +53,39 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun requestFasterRoute(
+        currentLocation: Location,
+        fasterRoutesRequestCallback: FasterRouteRequestCallback
+    ) {
+        if (routes.isEmpty()) {
+            return
+        }
+        routeOptions?.let { options ->
+            val origin = Point.fromLngLat(currentLocation.longitude, currentLocation.latitude)
+            val coordinates = options.coordinates().apply {
+                removeAt(0)
+                add(0, origin)
+            }
+            options.toBuilder().coordinates(coordinates)
```

Comment:
```suggestion
            val adjustedOptions = options.toBuilder().coordinates(coordinates).build()
```

This might not be enough though. We need to remove all waypoints that have already been visited, adjust their names, radiuses etc. Basically the same logic that needs to happen in
https://github.com/mapbox/mapbox-navigation-android/blob/83da15a0052645546da07b26beef9e8eaa1ec229/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L404-L453

just with a different request method. In this case, maybe it'd be better to compile new options in the `MapboxNavigation`?

Reply:
This is a good point. Will fix it